### DISCUSSION TITLE
Structured Configs and the Compose API

### DISCUSSION THREAD

**Discussion by jeppejohansen** [facebookresearch/hydra#2315] [Q&A]
I am currently using Hydra in conjunction with notebooks (or IPykernels to be specific) to configure a data science project.

At the moment I use the dictionary object when using the compose API. However, I would like to use the compose api and still be able to make a structured config. Is this currently a feature, and I just can find it in the documentation?

What i would like to be able to do:

```python

with initialize(config_path="../conf"):
    cfg = compose(config_name="config")
```

and having cfg being a structured config object?

**Comment by Jasha10** [comment]
Hi @JakartaLaw, sorry for the late reply.
Yes, you can use structured configs with the compose API.
You can follow the same steps for [structured config with the `hydra.main` API](https://hydra.cc/docs/1.2/tutorials/structured_config/intro/):
- Define some structured config dataclasses
- use the `ConfigStore.store` method to store your structured configs
- use the compose API to create a structured config object
